src/main/scala/viper/gobra/translator/implementations/components/OptionImpl.scala

```diff
@@ -0,0 +1,225 @@
+    vpr.DomainType(domainName, Map[vpr.TypeVar, vpr.Type]())(Seq(typeVar))
+  )(domainName = domainName)
+
+  /**
+    * Definition of the "optnone" function for the Viper domain of options:
+    *
+    * {{{
+    * function optnone() : Option[T]
+    * }}}
+    */
+  private lazy val optnone_func : vpr.DomainFunc = vpr.DomainFunc(
+    "optnone",
+    Seq(),
+    vpr.DomainType(domainName, Map[vpr.TypeVar, vpr.Type]())(Seq(typeVar))
+  )(domainName = domainName)
+
+  /**
+    * Definition of the "optget" function for the Viper domain of options:
+    *
+    * {{{
+    * function optget(o : Option[T]) : T
+    * }}}
+    */
+  private lazy val optget_func : vpr.DomainFunc = vpr.DomainFunc(
+    "optget",
+    Seq(vpr.LocalVarDecl("o", vpr.DomainType(domainName, Map[vpr.TypeVar, vpr.Type]())(Seq(typeVar)))()),
+    typeVar
+  )(domainName = domainName)
+
+  /**
+    * Definition of the "option_ex" axiom for the Viper domain of options:
+    *
+    * {{{
+    * axiom option_ex {
+    *   forall o : Option[T] :: o == optnone() || exists e : T :: o == optsome(e)
+    * }
+    * }}}
+    */
+  private lazy val option_ex_axiom : vpr.DomainAxiom = {
+    val oDecl = vpr.LocalVarDecl("o", vpr.DomainType(domainName, Map[vpr.TypeVar, vpr.Type]())(Seq(typeVar)))()
+    val eDecl = vpr.LocalVarDecl("e", typeVar)()
+
+    vpr.NamedDomainAxiom(
+      name = "option_ex",
+      exp = vpr.Forall(
+        Seq(oDecl),
+        Seq(),
+        vpr.Or(
+          vpr.EqCmp(oDecl.localVar, none(typeVar)())(),
+          vpr.Exists(
+            Seq(eDecl),
+            Seq(),
+            vpr.EqCmp(oDecl.localVar, some(eDecl.localVar)())()
+          )()
+        )()
+      )()
+    )(domainName = domainName)
+  }
+
+  /**
+    * Definition of the "optnone_some" axiom for the Viper domain of options:
+    *
+    * {{{
+    * axiom optnone_some {
+    *   forall e : T :: optsome(e) != optnone()
+    * }
+    * }}}
+    */
+  private lazy val optnone_some_axiom : vpr.DomainAxiom = {
+    val eDecl = vpr.LocalVarDecl("e", typeVar)()
+    val left = some(eDecl.localVar)()
+    val right = none(typeVar)()
+
+    vpr.NamedDomainAxiom(
+      name = "optnone_some",
+      exp = vpr.Forall(
+        Seq(eDecl),
+        Seq(),
+        vpr.NeCmp(left, right)()
+      )()
+    )(domainName = domainName)
+  }
+
+  /**
+    * Definition of the "optsome_over_get" axiom:
+    *
+    * {{{
+    * axiom optsome_over_get {
+    *   forall o : Option[T], e : T :: optget(o) == e <==> o == optsome(e)
+    * }
+    * }}}
+    */
+  private lazy val optsome_over_get_axiom : vpr.DomainAxiom = {
+    val oDecl = vpr.LocalVarDecl("o", vpr.DomainType(domainName, Map[vpr.TypeVar, vpr.Type]())(Seq(typeVar)))()
+    val eDecl = vpr.LocalVarDecl("e", typeVar)()
+    val left = get(oDecl.localVar, typeVar)()
+    val right = some(eDecl.localVar)()
+
+    vpr.NamedDomainAxiom(
+      name = "optsome_over_get",
+      exp = vpr.Forall(
+        Seq(oDecl, eDecl),
+        Seq(),
```

Comment:
This is the last time that I will write it, but all quantified expressions should get a trigger. When you want to use the default trigger, then instantiate it explicitly. We should always know what our triggers are.

Reply:
That sounds reasonable. I updated (most of) the axioms.